### PR TITLE
Add HA rolling_upgrade tests to SLE12 QAM

### PR DESCRIPTION
**Pull request by juadk**
This PR adds new rolling upgrade tests to SLE12 QAM.
_12-SP4 LTSS to 12-SP5
12-SP3 LTSS to 12-SP4 LTSS
12-SP2 LTSS to 12-SP3 LTSS_

- Related ticket: N/A
- Needles: N/A
- JIRA ticket: https://jira.suse.com/browse/TEAM-2101
- Verification run: 
12-SP4 LTSS to 12-SP5: [node1](http://1a102.qa.suse.de/tests/5850) - [node2](http://1a102.qa.suse.de/tests/5851)
12-SP3 LTSS to 12-SP4: [node1](http://1a102.qa.suse.de/tests/5847) - [node2](http://1a102.qa.suse.de/tests/5848)
12-SP2 LTSS to 12-SP3 LTSS: [node1](http://1a102.qa.suse.de/tests/5856) - [node2](http://1a102.qa.suse.de/tests/5857)

### CODE REVIEW

File: tests/migration/online_migration/register_without_ltss.pm [hunk 1]
```diff
@@ -28,6 +28,15 @@ sub run {
     set_var('SCC_ADDONS', join(',', @scc_addons));
 
     register_system_in_textmode;
+
```
Comment (juadk) [inline tests/migration/online_migration/register_without_ltss.pm:31]: We tried other solutions without adding a new setting but we agreed that the code is more readable with `LTSS_TO_LTSS`

File: tests/migration/online_migration/register_without_ltss.pm [hunk 2]
```diff
@@ -28,6 +28,15 @@ sub run {
...
+    # Sometimes in HA scenario, we need to test rolling upgrade migration from 
+    # a LTSS version to another one LTSS version.
+    # In this case, we need to set back ltss to SCC_ADDONS and set HDDVERSION to
+    # the targeted OS version for getting the good LTSS regcode.
+    if (get_var('LTSS_TO_LTSS')) {
```
Comment (juadk) [inline tests/migration/online_migration/register_without_ltss.pm:36]: We tried other solutions without adding a new setting but we agreed that the code is more readable with `LTSS_TO_LTSS`

File: tests/migration/online_migration/register_without_ltss.pm [hunk 3]
```diff
@@ -28,6 +28,15 @@ sub run {
...
+        set_var('SCC_ADDONS', join(',', @scc_addons, 'ltss'));
+        set_var('HDDVERSION', get_var('UPGRADE_TARGET_VERSION', get_var('VERSION')));
```
Comment (juadk) [inline tests/migration/online_migration/register_without_ltss.pm:38]: Needed for this [LTSS registration](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/registration.pm#L315-L319)

**Review by ricardobranco777 — APPROVED** [review]
LGTM

File: tests/migration/online_migration/register_without_ltss.pm [hunk 4]
```diff
@@ -28,6 +28,15 @@ sub run {
     set_var('SCC_ADDONS', join(',', @scc_addons));
 
     register_system_in_textmode;
+
+    # Sometimes in HA scenario, we need to test rolling upgrade migration from
+    # a LTSS version to another one LTSS version.
+    # In this case, we need to set back ltss to SCC_ADDONS and set HDDVERSION to
```
Comment (alvarocarvajald) [inline tests/migration/online_migration/register_without_ltss.pm:34]: I think `we need to add ltss again to SCC_ADDONS` is more clear

**Review by alvarocarvajald — APPROVED** [review]
LGTM. Just a small suggestion on the wording on one comment.

**Comment by alvarocarvajald** [comment]
BTW, congrats on this. Change is fairly limited, only a single mode and with the new setting, this should negatively impact any other tests. Nice job!

**Comment by juadk** [comment]
> BTW, congrats on this. Change is fairly limited, only a single module and with the new setting, this should not negatively impact any other tests. Nice job!
> 
> I think we can merge once the verification runs finish and travis' tests pass.

Thanks and you also participated in this as well, thanks for all your suggestions.

**Review by ldevulder — APPROVED** [review]
LGTM

**Pull request merged**